app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,40 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def new
+    @merchant = Merchant.new
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def create
+    merchant = Merchant.create!(merchant_params)
+    merchant.save
+    redirect_to "/merchants"
```

Comment:
you don't need to explicitly save this merchant. It will be saved once it is created.